src/components/CarOptions.js

```diff
@@ -1,9 +1,180 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { Button, Form, FormGroup, Label, Input, Col } from 'reactstrap';
+import FuelEconomyGov from '../data/FuelEconomyGov';
 
-const CarOptions = () => (
-    <div>
-        <h1>Car options</h1>
-    </div>
-)
+const FuelEconomy = new FuelEconomyGov();
 
-export default CarOptions;
+class DropdownMenu extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            currentYear: "",
+            make: [],
+            currentMake: "",
+            model: [],
+            currentModel: "",
+            option: [],
+            optionText: [],
+            currentOption: "",
+            id: [],
+            currentID: "",
+            currentCar: null,
+            optionState: null
+        };
+
+        this.yearChange = this.yearChange.bind(this);
+        this.makeChange = this.makeChange.bind(this);
+        this.modelChange = this.modelChange.bind(this);
+        this.optionChange = this.optionChange.bind(this);
+    }
+
+    yearChange(event) {
+        this.setState({
+            make: <option></option>,
+            model: <option></option>,
+            option: <option></option>
+        });
+        // console.log("Year selected")
+        this.setState({ currentYear: event.target.value }, () => {
+            let make = FuelEconomy.fetchMakesBy(this.state.currentYear);
+            make.unshift("")
+            this.setState({
+                make: make.map(make => {
+                    return (
+                        <option>
+                            {make}
+                        </option>
+                    );
+                })
+            })
+            // console.log(this.state.currentYear)
+        });
+    }
+
+    makeChange(event) {
+        this.setState({
+            model: <option></option>,
+            option: <option></option>
+        });
+        // console.log("Make selected")
+        this.setState({ currentMake: event.target.value }, () => {
+            let model = FuelEconomy.fetchModelsBy(this.state.currentYear, this.state.currentMake);
+            model.unshift("")
+            this.setState({
+                model: model.map(model => {
+                    return (
+                        <option>
+                            {model}
+                        </option>
+                    );
+                })
+            })
+            // console.log(this.state.currentMake)
+        });
+
+    }
+
+    modelChange(event) {
+        this.setState({
+            option: <option></option>
+        });
+        // console.log("Model selected")
+        this.setState({ currentModel: event.target.value }, () => {
```

Comment:
This code needs to be cleaned up. There's too many nested functions.

Reply:
I commented every function just for you baby<3